pymeasure/instruments/instrument.py

```diff
@@ -140,43 +141,53 @@ def control(get_command, set_command, docs,
         :param check_get_errors: Toggles checking errors after getting
         """
 
+        inverse = None
         if map_values and isinstance(values, dict):
             # Prepare the inverse values for performance
             inverse = {v: k for k, v in values.items()}
 
         def fget(self):
+            values_local = values
+            inverse_local = inverse
             vals = self.values(get_command, **kwargs)
             if check_get_errors:
                 self.check_errors()
+            if isinstance(values_local, str):
+                values_local = getattr(self, values_local)
             if len(vals) == 1:
                 value = get_process(vals[0])
                 if not map_values:
                     return value
-                elif isinstance(values, (list, tuple, range)):
-                    return values[int(value)]
-                elif isinstance(values, dict):
-                    return inverse[value]
+                elif isinstance(values_local, (list, tuple, range)):
+                    return values_local[int(value)]
+                elif isinstance(values_local, dict):
+                    if inverse_local is None:
+                        inverse_local = {v: k for k, v in values_local.items()}
+                    return inverse_local[value]
                 else:
                     raise ValueError(
                         'Values of type `{}` are not allowed '
-                        'for Instrument.control'.format(type(values))
+                        'for Instrument.control'.format(type(inverse_local))
```

Comment:
why the switch to inverse instead of values? It's the type of `values_local_ that gets checked in the ifs above.

Reply:
Oh good catch, that was a mistake indeed.